client/my-sites/domains/domain-management/list/item.jsx

```diff
@@ -30,21 +32,57 @@ class ListItem extends React.PureComponent {
 		onSelect: PropTypes.func.isRequired,
 		selectionIndex: PropTypes.number,
 		isSelected: PropTypes.bool,
+		shouldUpgradeToMakePrimary: PropTypes.bool.isRequired,
+		onUpgradeClick: PropTypes.func.isRequired,
 	};
 
+	renderContent() {
+		if ( this.props.enableSelection ) {
+			const content = <label htmlFor={ this.getInputId() }>{ this.content() }</label>;
+
+			if ( this.props.shouldUpgradeToMakePrimary ) {
+				return (
+					<div className="domain-management-list-item__content">
+						{ content }
+						{ this.upgradeToMakePrimary() }
+					</div>
+				);
+			}
+
+			return content;
+		}
+
+		return this.content();
+	}
+
 	render() {
+		const { busy, enableSelection, shouldUpgradeToMakePrimary } = this.props;
 		const cardClass = classNames( 'domain-management-list-item', {
-			busy: this.props.busy,
+			busy: busy || ( enableSelection && shouldUpgradeToMakePrimary ),
 		} );
 
 		return (
 			<CompactCard className={ cardClass } onClick={ this.handleClick }>
```

Comment:
I dug into the code for `CompactCard`/`Card` a bit, and the pointer on hover for the item is driven by the `is-clickable` class on the card, which is present when `props.onClick` is defined for the card. (See https://github.com/Automattic/wp-calypso/blob/master/packages/components/src/card/index.jsx#L43.)

Reply:
Thanks for tracking this down!